iceoryx_posh/source/popo/building_blocks/unique_port_id.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_posh/internal/popo/building_blocks/unique_port_id.hpp"
+#include "iceoryx_posh/iceoryx_posh_types.hpp"
+
+namespace iox
+{
+namespace popo
+{
+// in the future this should probably be moved to RouDi but currently it is only used for the ports
```

Comment:
Is this a specific class only to be used for RouDi and its ports or is it something generic? Please remove the comment if it will stay here.

Reply:
You are correct. It will stay here and also for a good reason. It is not generic but needed for the chunk header to set the origin id. This is in turn is then used for the ROS2 guid for publisher